t_frang/test_http_resp_code_block.py

```diff
@@ -0,0 +1,393 @@
+produce error HTTP responses.
+
+Tempesta FW provides http_resp_code_block for efficient blocking of all types of
+password crackers
+"""
+
+from requests import request
+from framework import tester
+
+__author__ = 'Tempesta Technologies, Inc.'
+__copyright__ = 'Copyright (C) 2019 Tempesta Technologies, Inc.'
+__license__ = 'GPL2'
+
+class HttpRespCodeBlockBase(tester.TempestaTest):
+    backends = [
+        {
+            'id' : 'nginx',
+            'type' : 'nginx',
+            'status_uri' : 'http://${server_ip}:8000/nginx_status',
+            'config' : """
+pid ${pid};
+worker_processes  auto;
+
+events {
+    worker_connections   1024;
+    use epoll;
+}
+
+http {
+    keepalive_timeout ${server_keepalive_timeout};
+    keepalive_requests 10;
+    sendfile         on;
+    tcp_nopush       on;
+    tcp_nodelay      on;
+
+    open_file_cache max=1000;
+    open_file_cache_valid 30s;
+    open_file_cache_min_uses 2;
+    open_file_cache_errors off;
+
+    # [ debug | info | notice | warn | error | crit | alert | emerg ]
+    # Fully disable log errors.
+    error_log /dev/null emerg;
+
+    # Disable access log altogether.
+    access_log off;
+
+    server {
+        listen        ${server_ip}:8000;
+
+        location /uri1 {
+            return 404;
+        }
+        location /uri2 {
+            return 200;
+        }
+        location /uri3 {
+            return 405;
+        }
+        location /nginx_status {
+            stub_status on;
+        }
+    }
+}
+""",
+        }
+    ]
+
+    clients = [
+        {
+            'id' : 'deproxy',
+            'type' : 'deproxy',
+            'addr' : "${tempesta_ip}",
+            'port' : '80',
+            'interface' : True,
+            'rps': 6
+        },
+        {
+            'id' : 'deproxy2',
+            'type' : 'deproxy',
+            'addr' : "${tempesta_ip}",
+            'port' : '80',
+            'interface' : True,
+            'rps': 5
+        }, 
+        {
+            'id' : 'deproxy3',
+            'type' : 'deproxy',
+            'addr' : "${tempesta_ip}",
+            'port' : '80',
+            'rps': 5
+        },
+        {
+            'id' : 'deproxy4',
+            'type' : 'deproxy',
+            'addr' : "${tempesta_ip}",
+            'port' : '80',
+            'rps': 5
+        }
+    ]
+
+
+
+
+
+class HttpRespCodeBlock(HttpRespCodeBlockBase):
+    """Blocks an attacker's IP address if a protected web application return
+    5 error responses with codes 404 within 2 seconds. This is 2,5 per second.
+    """
+    tempesta = {
+        'config' : """
+server ${server_ip}:8000;
+
+frang_limits {
+    http_resp_code_block 404 405 5 2;
+    ip_block on;
+}
+
+""",
+    }
+    def test_two_clients_block_ip(self):
+        """
+        Two clients to be blocked by ip for a total of 404 requests
+        
+        """
+        requests = "GET /uri1 HTTP/1.1\r\n" \
+                   "Host: localhost\r\n" \
+                   "\r\n" * 10
```

Comment:
10 requests from one client always return 200, another 10 requests always return 404. so this was my way of making sure that even if the second client sent good requests, it would still get blocked by id after 5 requests. probably I could do with 6 requests of each type